scripts/bfasst/compare_waveforms/File_Generation/waveform_generator.py

```diff
@@ -0,0 +1,40 @@
+import subprocess
+import time
+
+
+"""Uses IVerilog to create a VCD file for viewing waveforms"""
+
+
+def generate_VCD(paths, i):
+    # Generate wavefiles
+    subprocess.run(
+        [
+            "iverilog",
+            "-o",
+            str(paths["dsn"]),
+            str(paths["tb"][i]),
+            paths["file"][i],
+            str(paths["cells_sim"]),
+        ]
+    )
+    subprocess.run(["vvp", str(paths["dsn"])])
+    subprocess.run(["mv", "test.vcd", paths["temp_vcd"][i]])
+    paths["dsn"].unlink()
+
+    gtkwave(paths, i)
+
+
+"""Creates a waveform that only takes into account the inputs/outputs (totally ignores all internal components) and then
+takes the input/output graph and turns it into a VCD file to be compared against."""
```

Comment:
Have you looked into using the $dumpvars(1, MODULENAME) syntax to limit iverilog's output to top level signals?
http://referencedesigner.com/tutorials/verilog/verilog_62.php

Reply:
This would have been a life-saver when developing this script!!! Once I finish clearing up this PR, I'm going to take a look to see what that changes with the rest of the logic of the tool.